src/main/kotlin/fi/metatavu/metaform/server/controllers/MetaformController.kt

```diff
@@ -297,6 +314,106 @@ class MetaformController {
             }
     }
 
+    /**
+     * Deletes replies from forms that are marked as deleted
+     */
+    @Scheduled(every="\${metaforms.deletion.interval}", delayed = "\${metaforms.deletion.delay}")
+    fun scheduledMetaformDelete() {
+        val metaform = listDeletedMetaforms().firstOrNull() ?: return
```

Comment:
Listing all and then just picking first is quite expensive operation, limit results by maxResults instead